toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -803,12 +804,12 @@ func FormatSinglePartition(partDevPath string, partition configuration.Partition
 			fsType = "vfat"
 		}
 
-		mkfsArgs := []string{"-t", fsType}
+		mkfsArgs := []string{"--timeout", timeoutInSeconds, diskDevPath, "mkfs", "-t", fsType}
 		mkfsArgs = append(mkfsArgs, mkfsOptions...)
 		mkfsArgs = append(mkfsArgs, partDevPath)
 
 		err = retry.Run(func() error {
-			_, stderr, err := shell.Execute("mkfs", mkfsArgs...)
+			_, stderr, err := shell.Execute("flock", mkfsArgs...)
 			if err != nil {
 				logger.Log.Warnf("Failed to format partition using mkfs: %v", stderr)
```

Comment:
Could we also update the log messages to mention that now either `flock` or `mkfs` could have failed. I think it would be a useful hint for debugging purposes.

Reply:
The string `"Failed to format partition using mkfs"` is unique within the codebase. So, if we observe it in the logs, we will know the code failed at this location.

Reply:
My comment was more about helping the dev more quickly know what was run from the log itself without having to look at the code. Especially since not everyone is familiar with it or needs to be. If the error message is not `mkfs`-specific, it will get a bit confusing and you'll need to dive into the code, to figure out what's happening.

Reply:
Ok, fixed.